src/frontend/src/expr/type_inference/func.rs

```diff
@@ -816,6 +816,9 @@ fn build_type_derive_map() -> FuncSigMap {
     map.insert(E::ToTimestamp, vec![T::Float64], T::Timestampz);
     map.insert(E::AtTimeZone, vec![T::Timestamp, T::Varchar], T::Timestampz);
     map.insert(E::AtTimeZone, vec![T::Timestampz, T::Varchar], T::Timestamp);
+    map.insert(E::DateTrunc, vec![T::Varchar, T::Timestamp], T::Timestamp);
+    map.insert(E::DateTrunc, vec![T::Varchar, T::Timestampz], T::Timestampz);
```

Comment:
Actually, we should only allow `[T::Varchar, T::Timestampz, T::Varchar]` and forbid its binary variant:
> When the input value is of type timestamp with time zone, the truncation is performed with respect to a particular time zone; for example, truncation to day produces a value that is midnight in that zone. By default, truncation is done with respect to the current [TimeZone](https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-TIMEZONE) setting, but the optional time_zone argument can be provided to specify a different time zone.

As of now we do not support implicit TimeZone setting and require explicit zone parameter in all such operations. #5826